source/vision/visionHandlerExtensionPoints.py

```diff
@@ -4,21 +4,105 @@
 #Copyright (C) 2018-2019 NV Access Limited, Babbage B.V.
 
+"""Module containing extension points for vision enhancement providers.
+Consult the documentation of L{EventExtensionPoints} for more details.
+"""
+
 from extensionPoints import Action
 
+# Python 3 conversion note: convert this to a dataclass.
 class EventExtensionPoints:
+	"""Data class containing extension points that will be used to notify 
+	vision enhancement providers about occuring events, particularly for NVDA Objects.
+	Though an instance of this class is created when initializing the vision handler,
+	it should never be accessed from the vision handler directly.
+	Instead, vision enhancement providers should implement a "registerEventExtensionPoints" method,
+	taking an instance of this class as the only argument, performing registration to the extension points it is interested in.
+	For an example, see the vision.NVDAHighlighter module.
+	"""
+	#: Notifies a vision enhancement provider when an object property has changed.
+	#: This allows a vision enhancement provider to take an action
+	#: when one of the properties of an object has changed.
+	#: For example, a magnifier can track the magnified area of the screen to this object,
+	#: when the name of the object has changed.
+	#: Handlers are called with two  arguments.
+	#: @param obj: The object that received a property change.
+	#: @type obj: L{NVDAObjects.NVDAObject}
+	#: @param property: The object's property that changed, e.g. "name" or "description".
+	#: @type property: str
+	post_objectUpdate = None
+	#: Notifies a vision enhancement provider when the focused NVDAObject has changed.
+	#: This allows a vision enhancement provider to take an action when the focus changed.
+	#: For example, a magnifier can track the magnified area of the screen to this object.
+	#: Handlers are called with one argument.
+	#: @param obj: The object that received focus.
+	#: @type obj: L{NVDAObjects.NVDAObject}
 	post_focusChange = None
+	#: Notifies a vision enhancement provider when the foreground NVDAObject has changed.
+	#: This allows a vision enhancement provider to take an action
+	#: when another window takes the foreground.
+	#: For example, a magnifier can track the magnified area of the screen to this object.
+	#: Handlers are called with one argument.
+	#: @param obj: The object that became the foreground object.
+	#: @type obj: L{NVDAObjects.NVDAObject}
 	post_foregroundChange = None
-	post_caretMove = None
+	#: Notifies a vision enhancement provider when a physical caret has moved.
+	#: This allows a vision enhancement provider to take an action
+	#: when the caret moves in a window.
+	#: For example, a magnifier can track the magnified area of the screen to the new caret position.
+	#: Handlers are strongly encouraged to cache the new caret position,
+	#: and handle the pending caret update at the end of every core cycle using L{post_coreCycle},
+	#: unless tehy delegate caret change handling to a separate thread.

```

Comment:
minor typo: tehy -> they

Reply:
This typo looks like it has been copied to a few of these comments, a search and replace should fix this.